tidy3d/components/subpixel_spec.py

```diff
@@ -0,0 +1,129 @@
+# Defines specifications for subpixel averaging
+from __future__ import annotations
+
+import pydantic.v1 as pd
+from typing import Union
+
+from .base import Tidy3dBaseModel, cached_property
+from .types import TYPE_TAG_STR
+
+# Default Courant number reduction rate in PEC conformal's scheme
+DEFAULT_COURANT_REDUCTION_PEC_CONFORMAL = 0.3
+
+
+class AbstractSubpixelAveragingMethod(Tidy3dBaseModel):
+    """Base class defining how to handle material assignment on structure interfaces."""
+
+    @cached_property
+    def courant_ratio(self) -> float:
+        """The scaling ratio applied to Courant number so that the courant number
+        in the simulation is ``sim.courant * courant_ratio``.
+        """
+        return 1.0
+
+
+class Staircasing(AbstractSubpixelAveragingMethod):
+    """Apply staircasing scheme to material assignment of Yee grids on structure boundaries.
+
+    Note
+    ----
+    For PEC interface, the algorithm is based on:
+
+        A. Taflove and S. C. Hagness, "Computational electromagnetics: the
+        finite-difference time-domain method", Chapter 10.3 (2005).
+    """
+
+
+class VolumetricAveraging(AbstractSubpixelAveragingMethod):
+    """Apply volumetric averaging scheme to material properties of Yee grids on structure boundaries.
+    The material property is averaged in the volume surrounding the Yee grid.
```

Comment:
Does volumetric averaging simply average the permittivity between a metal and dielectric material with weights based on their relative volumes?

Reply:
That's right